src/processors/operation-processor.ts

```diff
@@ -36,11 +36,54 @@ export default class OperationProcessor<ResourceT extends Resource> {
 
     if (result !== undefined) {
       eagerLoadedData = await this.eagerLoad(op, result);
+      eagerLoadedData = await this.computeRelationshipProperties(op, eagerLoadedData);
     }
 
     return this.convertToResources(op, result, eagerLoadedData);
   }
 
+  async computeRelationshipProperties(op, eagerLoadedData) {
+    const baseResourceClass = await this.resourceFor(op.ref.type);
+    for (const relationship in eagerLoadedData) {
+      if (eagerLoadedData.hasOwnProperty(relationship)) {
+
+        const relationResourceClass = baseResourceClass.schema.relationships[relationship] && baseResourceClass.schema.relationships[relationship].type();
+        if (relationResourceClass) {
+          const includedRelationship = eagerLoadedData[relationship];
+          const resourceProcessor = await this.processorFor(relationResourceClass.type);
+
+          for (const includedRelationshipResource in includedRelationship["direct"]) {
```

Comment:
Can we type `includedRelationship` to access `direct` without the string literal? Also, to avoid repeating the expression all over this block, I'd rather store it in a variable:

```ts
const directRelations = includedRelationship.direct; // or ["direct"] if you must
```

Reply:
We could also probably have two lesser functions to handle each type of relationship and delegate these instructions. Something like `includeDirectRelationships()` and `includeNestedRelationships()`.